BiasAndFairnessModule/src/eval_engine/fairness_compass_engine.py

```
@@ -0,0 +1,334 @@
+                if y_scores is None:
+                    raise ValueError(f"Metric {metric_name} requires y_scores")
+                raw_result = metric_func(y_true, y_scores, sensitive_attr)
+            else:
+                raw_result = metric_func(y_true, y_pred, sensitive_attr)
+            
+            # Convert the result to a float using our utility function
+            from .metrics import convert_metric_to_float
+            return convert_metric_to_float(raw_result, metric_name)
+                
+        except KeyError:
+            raise KeyError(f"Metric '{metric_name}' not found. Available metrics: {list_metrics()}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)